src/components/organisms/LocalSearch.tsx

```diff
@@ -0,0 +1,979 @@
+      //   const matchDataContent = getMatchData(
+      //     changedContent,
+      //     regex,
+      //     LOCAL_MERGE_SAME_LINE_RESULTS_INTO_ONE
+      //   )
+      //   if (matchDataContent.length !== numFoundItems) {
+      //     shouldUpdateResults = true
+      //   }
+      // }
+      // if (!shouldUpdateResults) {
+      //   return
+      // }
+
+      // restore replace focus before updating results otherwise it might render
+      // out of focus because value don't have time to propagate
+      setFocusingReplace(false)
+      updateSearchResults(
+        searchValue,
+        regexSearch,
+        caseSensitiveSearch,
+        true,
+        false
+      )
+    }
+    codeMirror.on('change', onContentChange)
+    return () => {
+      codeMirror.off('change', onContentChange)
+    }
+  }, [
+    caseSensitiveSearch,
+    codeMirror,
+    getSearchRegex,
+    regexSearch,
+    searchValue,
+    updateSearchResults,
+  ])
+
+  useEffectOnce(() => {
+    updateSearchResults(searchValue, regexSearch, caseSensitiveSearch)
+    if (showReplace && searchValue.length !== 0) {
+      setFocusingReplace(true)
+    } else {
+      focusSearchTextAreaInput(searchValue.length, true)
+    }
+  })
+
+  /* todo: [komediruzecki-13/02/2021] Can this be improved without React.Component?
+   *  the issue is that we want to disable navigation and other buttons while updating search results
+   *  the operations pipeline would be:
+   *    1. Disable UI while updating results
+   *    2. Update search results (set selected item, number of found items, ...)
+   *    3. Enable UI
+   *  the solution used here is done with loading state
+   *  debounce is used to separate frequent request and on dependency list the
+   *  found loading state is updated, after that useEffect is
+   *  listening on dependency and updates the search results
+   *  (this way UI have time to update (disable buttons) and after search results are loaded)
+   */
+  useEffect(() => {
+    if (loadingResults) {
+      updateSearchResults(
+        searchValue,
+        regexSearch,
+        caseSensitiveSearch,
+        false,
+        false
+      )
+      if (onUpdateSearchOptions) {
+        onUpdateSearchOptions({
+          caseSensitiveSearch: caseSensitiveSearch,
+          regexSearch: regexSearch,
+        })
+      }
+      setLoadingResults(false)
+    }
+  }, [
+    caseSensitiveSearch,
+    loadingResults,
+    onUpdateSearchOptions,
+    regexSearch,
+    searchValue,
+    updateSearchResults,
+  ])
+
+  useDebounce(
+    () => {
+      setLoadingResults(true)
+    },
+    SEARCH_DEBOUNCE_TIMEOUT,
+    [searchValue, regexSearch, caseSensitiveSearch]
+  )
+
+  return (
+    <LocalSearchContainer>
+      <SearchResultItem>
+        <LocalSearchInputLeft>
+          <LocalSearchIcon numRows={textAreaRows}>
+            <Icon path={mdiMagnify} />
+          </LocalSearchIcon>
+          <textarea
+            onClick={() => focusSearchTextAreaInput(0)}
```

Comment:
Use a callback hook. This behavior is used in many places

Reply:
Actually added default value = 0 for this.
Since the call to this is already a callback hook